src/node/blockstorage.cpp

```diff
@@ -361,19 +362,32 @@ void BlockManager::FindFilesToPruneManual(
         chain.GetRole(), last_block_can_prune, count);
 }
 
-uint64_t BlockManager::GetPruneTargetForChainstate(const Chainstate& chain, ChainstateManager& chainman) const
+uint64_t BlockManager::GetPruneTargetForChainstate(const Chainstate& chain, ChainstateManager& chainman, const uint64_t current_disk_usage) const
 {
     const auto number_of_chainstates{chainman.GetAll().size()};
     const uint64_t min_overall_target{MIN_DISK_SPACE_FOR_BLOCK_FILES * number_of_chainstates};
     auto target = std::max(min_overall_target, GetPruneTarget());
     uint64_t target_boost{0};
     if (m_opts.prune_target_during_init > -1 && chainman.IsInitialBlockDownload()) {
-        if ((uint64_t)m_opts.prune_target_during_init <= target) {
+        uint64_t prune_target_temporarily{(uint64_t)m_opts.prune_target_during_init};
+        if (prune_target_temporarily <= target) {
             target = std::max(min_overall_target, (uint64_t)m_opts.prune_target_during_init);
         } else if (chain.GetRole() != ChainstateRole::ASSUMEDVALID) {
+            static constexpr uint64_t disk_space_buffer{100_MiB};
+            const auto disk_space_available{current_disk_usage + fs::space(m_opts.blocks_dir).available};
+            if (disk_space_available < prune_target_temporarily + disk_space_buffer) {
+                // If
+                if (disk_space_available <= target + disk_space_buffer) {
+                    // Simply ignore prune_target_during_init
+                    prune_target_temporarily = target;
+                } else {
+                    prune_target_temporarily = disk_space_available - disk_space_buffer;
+                }
```

Comment:
Maybe I got it wrong, but this should be the other way around? If enough diskspace is available for normal target -> use normal target, otherwise restrict target to what is available.
```suggestion
                if (disk_space_available >= target + disk_space_buffer) {
                    // Simply ignore prune_target_during_init
                    prune_target_temporarily = target;
                } else {
                    prune_target_temporarily = disk_space_available - disk_space_buffer;
                }
```